libusb/include/usbdriver.h

```diff
@@ -146,13 +167,20 @@ typedef struct {
 } usb_modeswitch_t;
 
 
+typedef struct {
+	bool deviceCreated;
+	oid_t dev;
+	char devPath[32];
+} usb_event_insertion_t;
+
+
 typedef struct usb_driver usb_driver_t;
 
 
 typedef int (*usb_completion_handler_t)(usb_driver_t *drv, usb_completion_t *completion, const char *data, size_t len);
 
 
-typedef int (*usb_insertion_handler_t)(usb_driver_t *drv, usb_devinfo_t *devinfo);
+typedef int (*usb_insertion_handler_t)(usb_driver_t *drv, usb_devinfo_t *devinfo, usb_event_insertion_t *event);
```

Comment:
Please clarify how this API works: is `event` optional (can it be NULL)? When exactly is `event` populated? Only when the function returns 0, or in all cases?

Reply:
Added a comment